offchain-modules/src/packages/handlers/tron.ts

```diff
@@ -0,0 +1,270 @@
+      permissionId: this.committee.permissionId,
+    });
+
+    const unsignedWithMemoTx = await this.tronWeb.transactionBuilder.addUpdateData(unsigned_tx, memo, 'utf8');
+
+    let signed_tx = unsignedWithMemoTx;
+    for (let i = 0; i < this.committee.keys.length; i++) {
+      signed_tx = await this.tronWeb.trx.multiSign(signed_tx, this.committee.keys[i]);
+    }
+
+    const broad_tx = await this.tronWeb.trx.broadcast(signed_tx);
+    return broad_tx.txid;
+  }
+
+  private async multiSignTransferTrc20(unlockRecord: TronUnlock) {
+    const from = this.committee.address;
+    const to = unlockRecord.tronRecipientAddress;
+    const amount = unlockRecord.amount;
+    const trc20ContractAddress = unlockRecord.asset;
+    const memo = unlockRecord.memo;
+
+    const options = {
+      permissionId: this.committee.permissionId,
+      feeLimit: 1000000,
+    };
+    const functionSelector = 'transfer(address,uint256)';
+    const params = [
+      { type: 'address', value: to },
+      { type: 'uint256', value: amount },
+    ];
+
+    const unsigned_tx = await this.tronWeb.transactionBuilder.triggerSmartContract(
+      trc20ContractAddress,
+      functionSelector,
+      options,
+      params,
+      from,
+    );
+    const unsignedWithMemoTx = await this.tronWeb.transactionBuilder.addUpdateData(
+      unsigned_tx.transaction,
+      memo,
+      'utf8',
+    );
+
+    let signed_tx = unsignedWithMemoTx;
+    for (let i = 0; i < this.committee.keys.length; i++) {
+      signed_tx = await this.tronWeb.trx.multiSign(signed_tx, this.committee.keys[i]);
+    }
+    const broad_tx = await this.tronWeb.trx.broadcast(signed_tx);
+    return broad_tx.txid;
+  }
+
+  // watch the eth_unlock table and handle the new unlock events
+  // send tx according to the data
+  async watchUnlockEvents(): Promise<void> {
+    while (true) {
+      logger.debug('flush pending tx to confirm');
+      const pendingRecords = await this.db.getTronUnlockRecords('pending');
+      for (let i = 0; i < pendingRecords.length; i++) {
+        // check tx is confirmed
+        pendingRecords[i].status = 'confirmed';
+      }
+      await this.db.saveTronUnlock(pendingRecords);
+
+      logger.debug('get new unlock events and send tx');
+      const unlockRecords = await this.db.getTronUnlockRecords('init');
+      for (let i = 0; i < unlockRecords.length; i++) {
+        let txid: string;
+        switch (unlockRecords[i].assetType) {
+          case 'trx':
+            txid = await this.multiSignTransferTrx(unlockRecords[i]);
+          case 'trc10':
+            txid = await this.multiSignTransferTrc10(unlockRecords[i]);
+          case 'trc20':
+            txid = await this.multiSignTransferTrc20(unlockRecords[i]);
+        }
+        unlockRecords[i].tronUnlockTxHash = txid;
+        unlockRecords[i].tronUnlockTxIndex = 0;
+        unlockRecords[i].status = 'pending';
```

Comment:
the status transition is confusing. pending -> confirmed -> pending.
should separate them.

Reply:
watch unlock events will do two operations:
- get 'pending' tx from tron_unlock and flush to 'success'
- get 'todo' tx from tron_unlock, execute the tx and flush to 'pending'